tests/system/test_client.py

```diff
@@ -394,7 +394,7 @@ def test_create_table_with_real_custom_policy(self):
         taxonomy_parent = f"projects/{Config.CLIENT.project}/locations/us"
 
         new_taxonomy = datacatalog_types.Taxonomy(
-            display_name="Custom test taxonomy",
+            display_name=unique_resource_id("Custom test taxonomy"),
```

Comment:
Given that the back-end generates a unique ID, why mangle the human-readable version?

Reply:
It turned out that this isn't the case, and `display_name` is used as an identifier - trying to create another taxonomy with the same display name results in a 409 Already Exists.

When such taxonomy was not cleaned up in one of the test runs (for some mysterious reason), the subsequent runs suddenly started failing on `master` (actually, on all branches).

Reply:
OK.  In that case, that is a pretty surpising use of `unique_resource_id`, which takes an argument, `delimiter` (not `prefix`).  It can in theory appear twice in the generated string.  A more usual idiom would be:
```suggestion
            display_name="Custom test taxonomy" + unique_resource_id(),
```

Reply:
(BTW, the back-end behavior is totally bogus).

Reply:
Yeah, I was surprised too, did not expect the _display_name_ to affect the resource identity. :confused: